### PR TITLE
Improve benchmark firmware organization

### PR DESCRIPTION
**Pull request by NoeTerrier**
Add a way to parametrize benchmark. The helpers for reading environment values have been factorized into a crate and can be now be called inside firmware.

Move benchmark firmware into firmware/benchmark/

Add write_csr benchmark into the CI.

### CODE REVIEW

File: config/qemu-virt-benchmark.toml [hunk 1]
```diff
@@ -19,3 +19,4 @@ enable = true
 time = true
 instruction = true
 nb_exits = true
+nb_iter=10
```
Comment (NoeTerrier) [inline config/qemu-virt-benchmark.toml:22]:
```suggestion
nb_iter = 10
```

File: firmware/benchmark/csr_write/main.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+use core::arch::asm;
+
+use config_helpers::parse_usize_or;
+use miralis_abi::{setup_firmware, success};
+
+setup_firmware!(main);
+
+fn main() -> ! {
+    let iterations = parse_usize_or(option_env!("MIRALIS_BENCHMARK_NB_ITER"), 1);
```
Comment (CharlyCst) [inline firmware/benchmark/csr_write/main.rs:12]: We will need to read this environment variable from multiple firmware images in the future, so we should probably factor it out in a single place. The `abi` crate is made for that, it defines functions for the firmware to interact with Miralis and other common utilities (such as initializing the stack, etc...)

File: runner/src/artifacts.rs [hunk 3]
```diff
@@ -124,7 +124,37 @@ pub fn locate_artifact(name: &str) -> Option<Artifact> {
...
+    if let Some(artifact) = external.get(name) {
+        return Some(artifact.clone());
+    }
+
+    // Could not find artifact
+    None
+}
+
+/// Check if one entry match the name
+fn find_artifact(firmware_path: &PathBuf, name: &str) -> Option<Artifact> {
```
Comment (CharlyCst) [inline runner/src/artifacts.rs:157]: Let's call it `find_artifact_source`. There are multiple kinds of artifacts (sources or binaries), here we only look for sources that we can compile.

**Review by CharlyCst — CHANGES_REQUESTED** [review]
I have just some small remarks but looks good otherwise, thanks!

**Review by CharlyCst — APPROVED** [review]
Thanks for the changes, look all good :)

**Pull request merged**